src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -498,6 +539,21 @@ class GroupReadsByUmi
     }
   }
 
+  /** When a minimum UMI length is specified, truncates all the UMIs to the length of the shortest UMI.  For the paired
+    * assigner, truncates the first UMI and second UMI sepeartely.*/
+  private def truncateUmis(umis: Seq[Umi]): Seq[Umi] = this.minUmiLength match {
+    case None => umis
+    case Some(length) =>
+      this.assigner match {
+        case _: PairedUmiAssigner =>
+          throw new IllegalStateException("Cannot used the paired umi assigner when min-umi-length is defined.")
+        case _ =>
+          val minLength = umis.map(_.length).min
+          require(length <= minLength, s"Bug: UMI found that had shorter length than expected ($minLength < $length)")
+          umis.map(_.take(minLength))
```

Comment:
FYI I think _probably_ it would be more efficient here to use `_.substring(0, minLength)` as that's a direct method on String and won't go through all the collections shenanigans.